speed-up the BaseDocumentApiV2Test

ivansenic opened the pull request:
So the speed-up for my machine is from ~10min to ~40sec:

* used `@ExtendWith(CqlSessionExtension.class) & @CqlSessionSpec()` to create sessions and keyspace only once
* fixed the keyspace name to constant
* in after test, don't drop the keyspace, but truncate the `collection` table (not sure if there is any other table created, I guess not)
* don't get auth token if we already have it..

Fun fact: `10 PRs checks a day x 3 builds x 260 working days x 9 mins savings = 4,212,000 seconds saved or ~$1322 a year for n1-highcpu-32 machine` :smile:

dimas-b commented on testing/src/main/java/io/stargate/it/http/docsapi/BaseDocumentApiV2Test.java:
```diff
@@ -52,43 +55,14 @@ public void setup(StargateConnectionInfo cluster) throws IOException {
-        .isTrue();
-
-    initAuth();
+    if (null == authToken) {
+      initAuth();
```
Could be done separately in a `@BeforeAll` method, I guess

ivansenic replied:
Done

dimas-b reviewed (approved):
I think this is a correct optimization and does not reduce test effectiveness/coverage. Nice catch! :)

ivansenic commented:
P.S. Quick overview on tests longer that 100s:

```
Step #4 - "cassandra-3.11": [WARNING] Tests run: 29, Failures: 0, Errors: 0, Skipped: 5, Time elapsed: 169.208 s - in io.stargate.db.cassandra.impl.CassandraPersistenceIT
Step #4 - "cassandra-3.11": [INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 101.16 s - in io.stargate.it.cql.SystemTablesTest
Step #4 - "cassandra-3.11": [INFO] Tests run: 13, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 189.437 s - in io.stargate.it.cql.JwtAuthTest
Step #4 - "cassandra-3.11": [INFO] Tests run: 29, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 309.986 s - in io.stargate.it.http.graphql.graphqlfirst.DataTypesTest
Step #4 - "cassandra-3.11": [INFO] Tests run: 45, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 130.997 s - in io.stargate.it.http.RestApiTest
Step #4 - "cassandra-3.11": [INFO] Tests run: 83, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 147.961 s - in io.stargate.it.http.RestApiv2Test
Step #4 - "cassandra-3.11": [INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 129.421 s - in io.stargate.it.http.MetricsTest
Step #4 - "cassandra-3.11": [INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 127.094 s - in io.stargate.it.http.RestApiTableTokenAuthTest
Step #4 - "cassandra-3.11": [WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 134.895 s - in io.stargate.it.http.docsapi.DocumentApiV2TestDisable2i
Step #4 - "cassandra-3.11": [INFO] Tests run: 49, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 676.012 s - in io.stargate.it.http.docsapi.DocumentApiV2Test
Step #4 - "cassandra-3.11": [INFO] Tests run: 49, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 673.135 s - in io.stargate.it.http.docsapi.BaseDocumentApiV2Test
```

edwardoliveira commented on testing/src/main/java/io/stargate/it/http/docsapi/BaseDocumentApiV2Test.java:
```diff
@@ -163,54 +137,54 @@ public void testUnauthorized() throws IOException {
     // Missing token header
     RestUtils.post(
-        null, hostWithPort + "/v2/namespaces/" + keyspace + "/collections/collection", data, 401);
+        null, hostWithPort + "/v2/namespaces/" + KEYSPACE + "/collections/collection", data, 401);
```
`Restv2ApiTest` uses `String.format` as below:

```
String.format("%s/v2/namespaces/%s/collections/collection", hostWithPort, KEYSPACE);
```

We can even extract `%s/v2/namespaces/%s/collections/collection` as a constant.

edwardoliveira commented on testing/src/main/java/io/stargate/it/http/docsapi/BaseDocumentApiV2Test.java:
```diff
@@ -163,54 +137,54 @@ public void testUnauthorized() throws IOException {
     // Missing token header
     RestUtils.post(
-        null, hostWithPort + "/v2/namespaces/" + keyspace + "/collections/collection", data, 401);
+        null, hostWithPort + "/v2/namespaces/" + KEYSPACE + "/collections/collection", data, 401);
```
Restv2ApiTest uses String.format as below:

String.format("%s/v2/namespaces/%s/collections/collection", hostWithPort, KEYSPACE);
We can even extract %s/v2/namespaces/%s/collections/collection as a constant.

ivansenic replied:
Yup, will do..

dimas-b commented:
@ivansenic : please review DSE test failures... Could they be related to the change in test setup/tearDown logic?

ivansenic commented:
> @ivansenic : please review DSE test failures... Could they be related to the change in test setup/tearDown logic?

I'll check if I missed something in the session creation that could cause this below. But imo this looks very suspicious. 

```
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:41,220 ProcessRunner.java:78 - stargate32-0> INFO  [CoreThread-18] 2021-05-12 17:05:41,220 QueryProcessor.java:426 - Executing DDL statement 'CREATE KEYSPACE IF NOT EXISTS "ks_1620839141218_BaseDocumentApiV2Test" WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1}' from client IP /127.0.0.1:42994 by user #<User anonymous>
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:42,631 ProcessRunner.java:78 - stargate32-0> INFO  [CoreThread-2] 2021-05-12 17:05:42,630 QueryProcessor.java:426 - Executing DDL statement 'create keyspace if not exists ks_docs_test WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1 }' from client IP /127.0.0.1:42994 by user #<User anonymous>
Step #4 - "dse-6.8": INFO  [main] 2021-05-12 17:05:42,632 ExternalResource.java:90 - Ignoring null ExternalStorage spec in [engine:junit-jupiter]/[class:io.stargate.it.http.docsapi.BaseDocumentApiV2Test]/[method:testBasicSearchSelectionSet()]
Step #4 - "dse-6.8": INFO  [main] 2021-05-12 17:05:42,632 ExternalResource.java:90 - Ignoring null StargateContainer spec in [engine:junit-jupiter]/[class:io.stargate.it.http.docsapi.BaseDocumentApiV2Test]/[method:testBasicSearchSelectionSet()]
Step #4 - "dse-6.8": INFO  [main] 2021-05-12 17:05:42,632 ExternalStorage.java:131 - About to run [engine:junit-jupiter]/[class:io.stargate.it.http.docsapi.BaseDocumentApiV2Test]/[method:testBasicSearchSelectionSet()] with storage cluster DSE 6.8
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:42,869 ProcessRunner.java:78 - stargate32-0> WARN  [mainIOThread-0] 2021-05-12 17:05:42,869 ClientState.java:339 - User 'cassandra' logged in from /127.0.0.1:0. It is strongly recommended to create and use another user and grant it superuser capabilities and remove the default one. See https://docs.datastax.com/en/dse/6.8/dse-admin/datastax_enterprise/security/Auth/secCreateRootAccount.html
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,295 ProcessRunner.java:78 - stargate32-0> ERROR [jersey-server-managed-async-executor-0] 2021-05-12 17:05:43,293 DocumentResourceV2.java:838 - Error when executing request
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> java.lang.IllegalArgumentException: Invalid value provided for 'bool_value': Java value true of type 'java.lang.Boolean' is not a valid value for CQL type tinyint
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.db.query.builder.QueryStringBuilder.valueToString(QueryStringBuilder.java:154)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.db.query.builder.QueryStringBuilder.append(QueryStringBuilder.java:142)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.db.query.builder.QueryStringBuilder.append(QueryStringBuilder.java:104)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.db.query.builder.QueryBuilderImpl.lambda$insertQuery$18(QueryBuilderImpl.java:1319)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.db.query.builder.QueryStringBuilder$ListBuilder.addAllWithIdx(QueryStringBuilder.java:233)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.db.query.builder.QueryBuilderImpl.insertQuery(QueryBuilderImpl.java:1314)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.db.query.builder.QueryBuilderImpl.build(QueryBuilderImpl.java:832)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.db.query.builder.QueryBuilder$QueryBuilder__22.build(QueryBuilder.java:1519)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.web.docsapi.dao.DocumentDB.getInsertStatement(DocumentDB.java:475)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.web.docsapi.dao.DocumentDB.deleteThenInsertBatch(DocumentDB.java:622)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.web.docsapi.service.DocumentService.putAtPath(DocumentService.java:443)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.web.docsapi.resources.DocumentResourceV2.lambda$putDoc$1(DocumentResourceV2.java:178)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.web.docsapi.resources.DocumentResourceV2.handle(DocumentResourceV2.java:808)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.web.docsapi.resources.DocumentResourceV2.putDoc(DocumentResourceV2.java:171)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at java.lang.reflect.Method.invoke(Method.java:498)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:124)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,296 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:167)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:176)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:79)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:475)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.server.model.ResourceMethodInvoker.lambda$apply$0(ResourceMethodInvoker.java:387)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.server.ServerRuntime$AsyncResponder$2$1.run(ServerRuntime.java:816)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at org.glassfish.jersey.server.ServerRuntime$AsyncResponder$2.run(ServerRuntime.java:811)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at java.lang.Thread.run(Thread.java:748)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> Caused by: java.lang.IllegalArgumentException: Java value true of type 'java.lang.Boolean' is not a valid value for CQL type tinyint
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.db.schema.Column.toCQLString(Column.java:90)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.db.schema.Column.access$000(Column.java:54)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.db.schema.Column$ColumnType.toCQLString(Column.java:191)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.db.query.builder.QueryStringBuilder.valueToString(QueryStringBuilder.java:151)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> 	... 37 common frames omitted
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,297 ProcessRunner.java:78 - stargate32-0> Caused by: io.stargate.db.schema.Column$ValidationException: Wanted 'tinyint' but got '' at location 'unknown'
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,298 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.db.schema.Column$ColumnType.validate(Column.java:173)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,298 ProcessRunner.java:78 - stargate32-0> 	at io.stargate.db.schema.Column.toCQLString(Column.java:86)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-12 17:05:43,298 ProcessRunner.java:78 - stargate32-0> 	... 40 common frames omitted
Step #4 - "dse-6.8": ERROR [main] 2021-05-12 17:05:43,324 RestUtils.java:285 - {"description":"Server error: Invalid value provided for 'bool_value': Java value true of type 'java.lang.Boolean' is not a valid value for CQL type tinyint","code":500}
Step #4 - "dse-6.8": WARN  [main] 2021-05-12 17:05:43,325 ExternalStorage.java:140 - Error in [engine:junit-jupiter]/[class:io.stargate.it.http.docsapi.BaseDocumentApiV2Test]/[method:testBasicSearchSelectionSet()]
Step #4 - "dse-6.8": org.opentest4j.AssertionFailedError: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <500>
Step #4 - "dse-6.8": to be equal to:
Step #4 - "dse-6.8":  <200>
Step #4 - "dse-6.8": but was not.
```

ivansenic commented:
The only suspicious thing here can be the same target keyspace between different tests (as it's sub-classes).. So I went for auto-created keyspace per class name..

I also introduced the `collectionPath` var and refactored/reformatted the whole code..

Let's see if build goes through now.

ivansenic commented:
OK the update helped, as now only 4 tests are failing.. Seems that this below si starting it and seems it's `DocumentApiV2TestDisable2i` test.. 

```
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,494 ProcessRunner.java:78 - storage0-0> INFO  [MiscStage:1] 2021-05-13 12:00:09,493  ColumnFamilyStore.java:2640 - Truncating ks_1620907191989_DocumentApiV2TestDisable2i.collection
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,565 ProcessRunner.java:78 - storage0-0> ERROR [MemtableFlushWriter:203] 2021-05-13 12:00:09,564  MemtableIndexWriter.java:122 - [ks_1620907191989_DocumentApiV2TestDisable2i.collection.collection_bool_value_idx] Error while flushing index new position should not be negative
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,565 ProcessRunner.java:78 - storage0-0> java.lang.IllegalArgumentException: new position should not be negative
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,565 ProcessRunner.java:78 - storage0-0> 	at org.apache.cassandra.io.util.RandomAccessReader.seek(RandomAccessReader.java:297)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,565 ProcessRunner.java:78 - storage0-0> 	at org.apache.cassandra.index.sai.disk.io.IndexInputReader.seek(IndexInputReader.java:150)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,565 ProcessRunner.java:78 - storage0-0> 	at org.apache.cassandra.index.sai.disk.v1.TraversingBKDReader.<init>(TraversingBKDReader.java:55)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,565 ProcessRunner.java:78 - storage0-0> 	at org.apache.cassandra.index.sai.disk.v1.NumericIndexWriter.writeAll(NumericIndexWriter.java:144)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,565 ProcessRunner.java:78 - storage0-0> 	at org.apache.cassandra.index.sai.disk.MemtableIndexWriter.flush(MemtableIndexWriter.java:151)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,565 ProcessRunner.java:78 - storage0-0> 	at org.apache.cassandra.index.sai.disk.MemtableIndexWriter.flush(MemtableIndexWriter.java:104)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,565 ProcessRunner.java:78 - storage0-0> 	at org.apache.cassandra.index.sai.disk.StorageAttachedIndexWriter.complete(StorageAttachedIndexWriter.java:184)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,565 ProcessRunner.java:78 - storage0-0> 	at java.util.ArrayList.forEach(ArrayList.java:1259)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,565 ProcessRunner.java:78 - storage0-0> 	at org.apache.cassandra.io.sstable.format.SSTableWriter.commit(SSTableWriter.java:278)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,565 ProcessRunner.java:78 - storage0-0> 	at org.apache.cassandra.io.sstable.SimpleSSTableMultiWriter.commit(SimpleSSTableMultiWriter.java:88)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,565 ProcessRunner.java:78 - storage0-0> 	at org.apache.cassandra.db.ColumnFamilyStore$Flush.flushMemtable(ColumnFamilyStore.java:1544)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,566 ProcessRunner.java:78 - storage0-0> 	at org.apache.cassandra.db.ColumnFamilyStore$Flush.run(ColumnFamilyStore.java:1429)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,566 ProcessRunner.java:78 - storage0-0> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,566 ProcessRunner.java:78 - storage0-0> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,566 ProcessRunner.java:78 - storage0-0> 	at org.apache.cassandra.concurrent.NamedThreadFactory.lambda$threadLocalDeallocator$0(NamedThreadFactory.java:88)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,566 ProcessRunner.java:78 - storage0-0> 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,566 ProcessRunner.java:78 - storage0-0> 	at java.lang.Thread.run(Thread.java:748)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:09,566 ProcessRunner.java:78 - storage0-0> 	at org.apache.cassandra.utils.concurrent.InlinedThreadLocalThread.run(InlinedThreadLocalThread.java:251)
```

And seems it's ending up with read failure (timeout) then:

```
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,780 ProcessRunner.java:78 - stargate30-0> ERROR [jersey-server-managed-async-executor-31] 2021-05-13 12:00:10,779 DocumentResourceV2.java:838 - Error when executing request
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,780 ProcessRunner.java:78 - stargate30-0> java.util.concurrent.CompletionException: org.apache.cassandra.stargate.exceptions.ReadFailureException: Operation failed - received 0 responses and 1 failures: TIMEOUT from 127.0.0.1:7000
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,780 ProcessRunner.java:78 - stargate30-0> 	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,780 ProcessRunner.java:78 - stargate30-0> 	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,780 ProcessRunner.java:78 - stargate30-0> 	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,780 ProcessRunner.java:78 - stargate30-0> 	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,780 ProcessRunner.java:78 - stargate30-0> 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,780 ProcessRunner.java:78 - stargate30-0> 	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,780 ProcessRunner.java:78 - stargate30-0> 	at io.stargate.db.dse.impl.DsePersistence$DseConnection.lambda$executeRequest$2(DsePersistence.java:557)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.observers.ConsumerSingleObserver.onError(ConsumerSingleObserver.java:46)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onError(SingleMap.java:69)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleFlatMap$SingleFlatMapCallback$FlatMapSingleObserver.onError(SingleFlatMap.java:116)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleFlatMap$SingleFlatMapCallback.onError(SingleFlatMap.java:90)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onError(SingleMap.java:69)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleFlatMap$SingleFlatMapCallback$FlatMapSingleObserver.onError(SingleFlatMap.java:116)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.observers.ResumeSingleObserver.onError(ResumeSingleObserver.java:51)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.disposables.EmptyDisposable.error(EmptyDisposable.java:78)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleError.subscribeActual(SingleError.java:42)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.Single.subscribe(Single.java:3603)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleDelayWithCompletable$OtherObserver.onComplete(SingleDelayWithCompletable.java:69)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.disposables.EmptyDisposable.complete(EmptyDisposable.java:68)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.completable.CompletableEmpty.subscribeActual(CompletableEmpty.java:27)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.Completable.subscribe(Completable.java:2309)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleDelayWithCompletable.subscribeActual(SingleDelayWithCompletable.java:36)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.Single.subscribe(Single.java:3603)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleDefer.subscribeActual(SingleDefer.java:43)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.Single.subscribe(Single.java:3603)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at org.apache.cassandra.utils.flow.RxThreads$1SubscribeOn.lambda$subscribeActual$0(RxThreads.java:51)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at org.apache.cassandra.concurrent.TPCRunnable.run(TPCRunnable.java:101)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at org.apache.cassandra.concurrent.StagedScheduler.execute(StagedScheduler.java:98)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at org.apache.cassandra.utils.flow.RxThreads$1SubscribeOn.subscribeActual(RxThreads.java:51)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.Single.subscribe(Single.java:3603)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,781 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleFlatMap$SingleFlatMapCallback.onSuccess(SingleFlatMap.java:84)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleJust.subscribeActual(SingleJust.java:30)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.Single.subscribe(Single.java:3603)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleFlatMap.subscribeActual(SingleFlatMap.java:36)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.Single.subscribe(Single.java:3603)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.Single.subscribe(Single.java:3603)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleFlatMap.subscribeActual(SingleFlatMap.java:36)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.Single.subscribe(Single.java:3603)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.internal.operators.single.SingleDefer.subscribeActual(SingleDefer.java:43)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.reactivex.Single.subscribe(Single.java:3603)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at org.apache.cassandra.utils.flow.RxThreads$1SubscribeOn.lambda$subscribeActual$0(RxThreads.java:51)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at org.apache.cassandra.concurrent.TPCRunnable.run(TPCRunnable.java:101)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:465)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at java.lang.Thread.run(Thread.java:748)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at org.apache.cassandra.utils.concurrent.InlinedThreadLocalThread.run(InlinedThreadLocalThread.java:251)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> Caused by: org.apache.cassandra.stargate.exceptions.ReadFailureException: Operation failed - received 0 responses and 1 failures: TIMEOUT from 127.0.0.1:7000
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.stargate.db.dse.impl.Conversion.toExternal(Conversion.java:373)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.stargate.db.dse.impl.Conversion.convertInternalException(Conversion.java:548)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.stargate.db.dse.impl.DsePersistence$DseConnection.convertExceptionWithWarnings(DsePersistence.java:568)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	at io.stargate.db.dse.impl.DsePersistence$DseConnection.lambda$executeRequest$2(DsePersistence.java:555)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,782 ProcessRunner.java:78 - stargate30-0> 	... 43 common frames omitted
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 	Suppressed: org.apache.cassandra.exceptions.ReadFailureException: Operation failed - received 0 responses and 1 failures (INDEX_NOT_AVAILABLE=[/127.0.0.1])
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.index.SecondaryIndexManager.filterForQuery(SecondaryIndexManager.java:445)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.service.reads.range.RangeIterator.computeNext(RangeIterator.java:83)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.service.reads.range.RangeIterator.computeNext(RangeIterator.java:47)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.utils.AbstractIterator.hasNext(AbstractIterator.java:50)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at com.google.common.collect.Iterators$PeekingImpl.hasNext(Iterators.java:1171)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.service.reads.range.RangeMerger.hasNext(RangeMerger.java:62)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.service.reads.range.EndpointGroupingCoordinator.<init>(EndpointGroupingCoordinator.java:72)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.service.reads.range.EndpointGroupingRangeReadExecutor.sendNextRequests(EndpointGroupingRangeReadExecutor.java:90)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.service.reads.range.RangeReadExecutor.nextBatch(RangeReadExecutor.java:240)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.service.reads.range.RangeReadExecutor.partitions(RangeReadExecutor.java:199)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.service.StorageProxy.getRangeSliceRemote(StorageProxy.java:1535)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.service.StorageProxy.getRangeSlice(StorageProxy.java:1527)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.db.PartitionRangeReadCommand.execute(PartitionRangeReadCommand.java:360)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.service.pager.AbstractQueryPager.lambda$fetchPage$0(AbstractQueryPager.java:90)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.service.pager.AbstractQueryPager.innerFetch(AbstractQueryPager.java:130)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.service.pager.AbstractQueryPager.fetchPage(AbstractQueryPager.java:90)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.service.pager.PartitionRangeQueryPager.fetchPage(PartitionRangeQueryPager.java:31)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.cql3.statements.SelectStatement$Pager$NormalPager.fetchPage(SelectStatement.java:472)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.cql3.statements.SelectStatement.execute(SelectStatement.java:523)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.cql3.statements.SelectStatement.execute(SelectStatement.java:620)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.cql3.statements.SelectStatement.execute(SelectStatement.java:348)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at org.apache.cassandra.cql3.QueryProcessor.lambda$processStatementImpl$4(QueryProcessor.java:453)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		at io.reactivex.internal.operators.single.SingleDefer.subscribeActual(SingleDefer.java:36)
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,783 ProcessRunner.java:78 - stargate30-0> 		... 26 common frames omitted
Step #4 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-05-13 12:00:10,784 ProcessRunner.java:78 - stargate30-0> INFO  [jersey-server-managed-async-executor-31] 2021-05-13 12:00:10,781 Slf4jRequestLogWriter.java:62 - 127.0.0.1 - - [13/May/2021:12:00:10 +0000] "GET /v2/namespaces/ks_1620907191989_DocumentApiV2TestDisable2i/collections/collection HTTP/1.1" 500 189 "-" "okhttp/4.9.0" 18
Step #4 - "dse-6.8": ERROR [main] 2021-05-13 12:00:10,785 RestUtils.java:285 - {"description":"Server error: org.apache.cassandra.stargate.exceptions.ReadFailureException: Operation failed - received 0 responses and 1 failures: TIMEOUT from 127.0.0.1:7000","code":500}
Step #4 - "dse-6.8": WARN  [main] 2021-05-13 12:00:10,788 ExternalStorage.java:140 - Error in [engine:junit-jupiter]/[class:io.stargate.it.http.docsapi.DocumentApiV2TestDisable2i]/[method:testGetFullDocMultiFilter()]
Step #4 - "dse-6.8": org.opentest4j.AssertionFailedError: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <500>
Step #4 - "dse-6.8": to be equal to:
Step #4 - "dse-6.8":  <200>
```

Ideas?

ivansenic commented:
OK seems that this problem with SAI indexes where there before as well.. They were happening on dropping the keyspace used in the single test.. However, tests were green as we were creating a new keyspace for every test method, thus it was not affecting future tests as now.

This however, smells to a way bigger problem. Maybe in case of SAI we are not defining the indexes in a correct way and thus we have a problem atm. Not sure, but this seems happening on both truncate table and delete keyspace. Why would this be a case? Also it seems to be happening only after certain tests:
* `testPutReplacingWithArray` 
* `testRootDocumentPatch`

@EricBorczuk @dimas-b This is smelling a lot. Can we join forces?

dimas-b commented:
For tests that use SAI we probably _have_ to use a different table name because their schema is supposed to be different.

ivansenic commented:
> For tests that use SAI we probably _have_ to use a different table name because their schema is supposed to be different.

Don't think so, each test gets different keyspace, so `DocumentApiV2Test` `DocumentApiV2TestDisable2i` would end up in different keyspaces..

ivansenic commented:
Trying with avoiding truncate:

```
  @AfterEach
  public void emptyTargetCollection(CqlSession session) {
    ResultSet results = session.execute(String.format("SELECT key FROM %s", TARGET_COLLECTION));
    List<String> ids = new ArrayList<>();
    results.forEach(row -> ids.add(String.format("'%s'", row.getString("key"))));

    if (!ids.isEmpty()) {
      session.execute(
          String.format(
              "DELETE FROM %s WHERE key IN (%s)", TARGET_COLLECTION, String.join(", ", ids)));
    }
  }
```

Tests are even faster now, under 20s on my machine..

ivansenic commented:
Merging as tests are green. Seems that SAI indexes are only throwing exception on the `TRUNCATE`, but not `DELETE`. :partying_face: 

@dimas-b @EricBorczuk Can you point me where to create the SAI index ticket for the DSE?
